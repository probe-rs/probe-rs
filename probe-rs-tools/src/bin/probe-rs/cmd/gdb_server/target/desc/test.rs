use probe_rs::{CoreType, InstructionSet};

use super::TargetDescription;

#[test]
fn test_target_description_microbit() {
    let target_desc = TargetDescription::new(CoreType::Armv6m, InstructionSet::Thumb2);
    let description = target_desc.get_target_xml();

    insta::assert_snapshot!(description);
}

#[test]
fn test_target_with_features() {
    let mut target_desc = TargetDescription::new(CoreType::Armv6m, InstructionSet::Thumb2);
    target_desc.add_gdb_feature("org.probe-rs.feature1");
    target_desc.add_register_from_details("r0", 32, 0.into());
    target_desc.add_register_from_details("x1", 64, 1.into());
    target_desc.add_register_from_details("t2", 64, 2.into());

    target_desc.update_register_name("t2", "at2");
    target_desc.update_register_type("at2", "special_reg");

    target_desc.add_gdb_feature("org.probe-rs.feature2");
    target_desc.add_register_from_details("v4", 128, 4.into());

    let description = target_desc.get_target_xml();

    insta::assert_snapshot!(description);
}
