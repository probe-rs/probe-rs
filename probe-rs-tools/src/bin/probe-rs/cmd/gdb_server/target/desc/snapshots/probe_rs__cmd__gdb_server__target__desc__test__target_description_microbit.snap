---
source: probe-rs-tools/src/bin/probe-rs/cmd/gdb_server/target/desc/test.rs
assertion_line: 10
expression: description
---
<?xml version="1.0"?>
        <!DOCTYPE target SYSTEM "gdb-target.dtd">
        <target version="1.0">
        <architecture>armv6-m</architecture></target>
